File: Development/WoodLotFrontEnd/src/app/entita/utente/utente.ts
import {Ordine} from "../ordine/ordine";
import {UtenteRegistrato} from "../utenteRegistrato/utente-registrato";


export class Utente extends UtenteRegistrato {

  public nome: String
  public cognome: String
  public listaOrdini: Ordine[]
  public id ?: number


  constructor(nome: String, cognome: String, listaOrdini: Ordine[], email: String, password: String, id?: number) {
    super(email, password)
    this.nome = nome
    this.cognome = cognome
    this.listaOrdini = listaOrdini
    this.email = email
    this.password = password
    this.id = id
  }
}

File: Development/WoodLotBackEnd/src/main/java/it/unisa/WoodLot/model/entity/Utente.java
package it.unisa.WoodLot.model.entity;

import lombok.Data;

import javax.persistence.*;

/**
 * Classe che modella un utente
 *
 * @author Alessia Ture
 */
@Data
@Entity
@Table(name = "utente")
public class Utente extends UtenteRegistrato {
    @Column(name = "nome", nullable = false)
    private String nome;

    @Column(name = "cognome", nullable = false)
    private String cognome;

    public Utente() {

    }

    public Utente(Long id, String email, String password, String nome, String cognome) {
        super(id, email, password);
        this.nome = nome;
        this.cognome = cognome;
    }

}
